### PR TITLE
[RPA-629] Changes in process of definition ExclusiveGateway

### PR DESCRIPTION
**Pull request by damlit**
## Description
Better user experience during writing expressions in exclusive gateway/flows.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import { ExpressionTextField, FlowExpression, GatewayFormMenu } from './GatewayFormRenderer.styles';
+
+import { BpmnConnectionFactory, IBpmnConnection, IBpmnGateway } from '../../helpers/elementParameters';
+
+import FlowLabelForm from '../FlowLabelForm';
+
+
+const GatewayFormRenderer = () => {
+    const COLOR_RB = '#FBB040';
+    const COLOR_BLACK = '#000000';
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: isnt it available in our theme?

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: imo use useTheme hook to get it

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: default theme object is here https://mui.com/material-ui/customization/default-theme/

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    
+    const { selectedElement } = useSelector(state => state.process.modeler);
+    const { modeler } = useModelerContext();
+    const gateway: IBpmnGateway = selectedElement as IBpmnGateway;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: type it only in 1 place

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    const [defaultFlow, setDefaultFlow] = React.useState(gateway.businessObject.default?.id);
+    
+    const createInitExpressions = () => {
+        const expressionsArray = gateway.outgoing
+            .map((flow) => ({
+                [flow.id]: flow.businessObject.conditionExpression?.body
+            }));
+        return Object.assign({}, ...expressionsArray);
+    };
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: use `reduce` here to convert list to object

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            <Grid item xs={12}>
+                <Box px={2} pt={1}>
+                    <Typography variant="h4" gutterBottom>
+                        {gateway.id}
+                    </Typography>
+                </Box>
+            </Grid>
+            <GatewayFormMenu>
+                <FormControl fullWidth>
+                    <InputLabel id="default-flow-select-input-label">Default flow</InputLabel>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: translation

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                    <Select
+                        labelId="default-flow-select-label"
+                        id="default-flow-select"
+                        value={defaultFlow}
+                        label="Default"
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: translation

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/FlowLabelForm.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    const handleCloseForm = () => {
+        setFormState({
+            editing: false,
+            label: selectedElement.businessObject.name as string,
+        });
+        onCancel();
+    };
+
+    const EditButtons = () => {
+        const isSameAsBefore =
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/FlowLabelForm.tsx]: `isEqual` imo better, avoid `as` and so on

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx (hunk 5 above)
Comment (damlit) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/renderers/GatewayFormRenderer.tsx]: Unnecessary label - removed

**Review by preludium — APPROVED** [review]

**Pull request merged**